test/jdk/java/foreign/StdLibTest.java

```diff
@@ -62,6 +62,14 @@ public class StdLibTest extends NativeTestHelper {
 
     final static Charset nativeCharset = Charset.forName(System.getProperty("native.encoding"));
 
+    static MemorySegment allocateSegmentForString(Arena arena, String str) {
+        try {
+            return arena.allocateFrom(str, nativeCharset);
+        } catch (IllegalArgumentException ex) {
+            return arena.allocateFrom(str);
+        }
```

Comment:
Why should this be restricted to standard charsets? Why not:
```java
        byte[] data = str.getBytes(nativeCharset);
        data = Arrays.copyOf(data, data.length + 1); // NUL-terminate.
        return arena.allocateFrom(ValueLayout.JAVA_BYTE, data);
```

Reply:
You are right, Looking at the API it should be actually right thing to do rather than going through the default charset. 
Code changed in https://github.com/ibmruntimes/openj9-openjdk-jdk/compare/d028c2086dd022dc7efe806a9a2d53f7d23c4540..6c2a1a01939d74c924fa89f0bcf96628c9c0b444

Pushing updated commit message as well.